### PR TITLE
Basemats 2: Base Meats, Wood datum, and Tile support

### LINKED ISSUE #10353: [Bug]: Wooden tables are not made of wood anymore
### BYOND Version

_No response_

### Round Date

2023/12/19

### Round ID

47038

### Test merges

_No response_

### Describe the bug

When attempting to create wooden tables via applying wooden planks to a table frame, I get a weird iron table instead that drops no items (save iron rods) when broken.

### What did you expect?

A lovely wooden table, not an ugly iron one.

### Reproducible steps

1. Make table frame
2. Apply wood
3. Get broken (iron?) table

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/48820
- https://github.com/tgstation/tgstation/pull/49402
- https://github.com/tgstation/tgstation/pull/50203
- https://github.com/tgstation/tgstation/pull/54926
- https://github.com/tgstation/tgstation/pull/55790

## About The Pull Request

closes #10353 

This PR adds quite the expansion to basemats. After this is merged, we will be about halfway through from a complete system (yay!)

### Basemats

The Basemat system overall has been expanded to support tiles and walls. They are constructed in the same way they were before through the crafting menu and using materials on girders, respectively.

### New Additions 

Of the additions, theres two new material datums in the form of Meat and Wood! Meat materials are made by putting raw meat into a specialized machine, wood is the same as before. As with all basemats, these can be inserted into the autolathe.

The aforementioned specialized machine is called the 'sheetifier'. It only has one usecase so far, but I'm looking to increase its functionality in the future.

### TODO

- [x] Ensure wooden tiles, walls, chairs, and tables
- [x] Ensure meat tiles, walls, chairs, and tables

## Why It's Good For The Game

Aforementioned fix for making wooden tables using iron rods.

I think this is good because it expands the basemat system, and also gives more room for future materials to be added. I received positive feedback for this on the discord, so I think it will be fun.

As with all basemats, items made of wood inherits the properties of the material they are made of. Many more wooden items should now be flammable. You can eat things made out of meat.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/6cf0e2cf-2d85-4d81-af04-c205979710c5

</details>

## Changelog
:cl: rkz, ArcaneMusic, qustinnus, 4Dplanner, cacogen, AnturK
add: Adds meat & wood material datums
add: Sheetifier machine, for making raw meat into a building material!
add: tile & walls support for the basemat system
tweak: more wooden items are now flammable
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
Mixed on the sheetifier overall. I dislike that it only has one usecase. Open to alternatives if you have them.

**Comment by Tsar-Salat** [comment]
shes reaaaaady!

File: code/game/objects/items/stacks/sheets/miscellaneous/miscellaneous_mats.dm [hunk 1]
```diff
@@ -147,5 +147,4 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
 	material_type = /datum/material/meat
 	material_modifier = 1 //None of that wussy stuff
 
-/obj/item/stack/sheet/meat/get_recipes()
-	return SSmaterials.rigid_stack_recipes
+/obj/item/stack/sheet/meat/get_recipes() //empty typepath. Leave it be.
```
Comment (EvilDragonfiend) [inline code/game/objects/items/stacks/sheets/miscellaneous/miscellaneous_mats.dm]: I guess you missed my answer in Discord..?
You need to return a recipe glob that's dedicated for this like how all other recipes do.

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/sheets/miscellaneous/miscellaneous_mats.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/game/objects/items/stacks/sheets/miscellaneous/miscellaneous_mats.dm]: fixed in 1ce8eb7

File: code/game/turfs/closed/wall/material_walls.dm [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_MATERIAL_WALLS, SMOOTH_GROUP_WALLS)
+	canSmoothWith = list(SMOOTH_GROUP_MATERIAL_WALLS)
+	material_flags = MATERIAL_EFFECTS | MATERIAL_ADD_PREFIX | MATERIAL_COLOR | MATERIAL_AFFECT_STATISTICS
+
+/turf/closed/wall/material/break_wall()
+	for(var/i in custom_materials)
+		var/datum/material/M = i
+		new M.sheet_type(src, FLOOR(custom_materials[M] / MINERAL_MATERIAL_AMOUNT, 1))
+	return new girder_type(src)
```
Comment (EvilDragonfiend) [inline code/game/turfs/closed/wall/material_walls.dm]:
```suggestion
	if(girder_type)
		return new girder_type(src)
```

File: code/game/turfs/closed/wall/mineral_walls.dm [hunk 3]
```diff
@@ -171,6 +171,7 @@
 	smoothing_flags = SMOOTH_BITMASK
 	smoothing_groups = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_WOOD_WALLS)
 	canSmoothWith = list(SMOOTH_GROUP_WOOD_WALLS)
+	custom_materials = list(/datum/material/wood = 4000)
```
Comment (EvilDragonfiend) [inline code/game/turfs/closed/wall/mineral_walls.dm:174]: change every `= 4000` into `MINERAL_MATERIAL_AMOUNT*2`
There are a few ones

**Review by EvilDragonfiend — APPROVED** [review]
Looks good to me. I see some values are not using define macros, but not really an issue here.

**Pull request merged**